### PR TITLE
AM2R: add option to unlock lab doors

### PR DESCRIPTION
**Pull request by Miepee**

### CODE REVIEW

File: randovania/gui/ui_files/preset_am2r_patches.ui [hunk 1]
```diff
@@ -213,6 +213,27 @@
...
+            <widget class="QCheckBox" name="force_blue_labs_check">
+             <property name="text">
+              <string>Unlock Genetics Laboratory Doors</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QLabel" name="force_blue_labs_check_2">
+             <property name="text">
+              <string>Ensures that the doors in the later parts of Genetics the are normal (blue) doors, even with door lock rando enabled.
```
Comment (ThanatosGit) [inline randovania/gui/ui_files/preset_am2r_patches.ui:231]: > [...] Genetics `the` are [...]
Remove `the` ?

File: test/games/am2r/generator/test_am2r_base_patches.py
Comment (ThanatosGit) [inline test/games/am2r/generator/test_am2r_base_patches.py:1]: 🐶 reported some stuff here.

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

File: randovania/gui/ui_files/preset_am2r_patches.ui (hunk 1 above)
Comment (Miepee) [reply randovania/gui/ui_files/preset_am2r_patches.ui:231]: multiple words are missing there, absolutely no idea how that happened

File: test/games/am2r/generator/test_am2r_base_patches.py
Comment (Miepee) [reply test/games/am2r/generator/test_am2r_base_patches.py:1]: think i managed to fix it

File: test/games/am2r/generator/test_am2r_base_patches.py [hunk 2]
```diff
@@ -4,54 +4,94 @@
...
-    1614,
-    1561,
-    1720,
-    1656,
-    601,
-]
-
-
-def test_base_patches(am2r_game_description, preset_manager) -> None:
+_save_door_mapping = {
```
Comment (ThanatosGit) [inline test/games/am2r/generator/test_am2r_base_patches.py:14]: Is it planned to have other values here? "Normal Door" everywhere seems very static.

File: test/games/am2r/generator/test_am2r_base_patches.py [hunk 3]
```diff
@@ -4,54 +4,94 @@
...
+    1796: "Normal Door",
+    1797: "Normal Door",
+    1804: "Normal Door",
+    1805: "Normal Door",
+    1808: "Normal Door",
+    1889: "Normal Door",
+    1894: "Normal Door",
+}
+
+_lab_door_mapping: dict = {}
```
Comment (ThanatosGit) [inline test/games/am2r/generator/test_am2r_base_patches.py:59]: This thing seems to be only referenced one time, which is when iterating over it in Line 94.

File: test/games/am2r/generator/test_am2r_base_patches.py [hunk 4]
```diff
@@ -4,54 +4,94 @@
...
+        for num, value in _save_door_mapping.items():
+            weakness = result.dock_weakness[num]
+            assert isinstance(weakness, DockWeakness)
+            assert weakness.name == value
+
+    if force_blue_labs:
+        for num, value in _lab_door_mapping.items():
+            weakness = result.dock_weakness[num]
+            assert isinstance(weakness, DockWeakness)
+            assert weakness.name == value
```
Comment (ThanatosGit) [inline test/games/am2r/generator/test_am2r_base_patches.py:97]: If `_lab_door_mapping` is always empty, this checks nothing.

File: randovania/gui/ui_files/preset_am2r_patches.ui [hunk 5]
```diff
@@ -213,6 +213,27 @@
...
+             <property name="text">
+              <string>Unlock Genetics Laboratory Doors</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QLabel" name="force_blue_labs_check_2">
+             <property name="text">
+              <string>Ensures that the doors in the later parts of Genetics Laboratory are normal (blue) doors, even with door lock rando enabled.
+To be more precise, all rooms after &quot;Hatchling Room Underside&quot; will have their doors changed.</string>
```
Comment (ThanatosGit) [inline randovania/gui/ui_files/preset_am2r_patches.ui:232]: A small thing here: Do we write region names without quotes but room names with quotes? Looks a litle bit mixed but dunno how other games do it.

Comment (Miepee) [reply randovania/gui/ui_files/preset_am2r_patches.ui:232]: honestly, no idea. i think each game just kinda does their own.

File: test/games/am2r/generator/test_am2r_base_patches.py (hunk 2 above)
Comment (Miepee) [reply test/games/am2r/generator/test_am2r_base_patches.py]: no plans to have other values, its static. removed this mapping, and reverted it back to a list of indices

File: test/games/am2r/generator/test_am2r_base_patches.py (hunk 4 above)
Comment (Miepee) [reply test/games/am2r/generator/test_am2r_base_patches.py]: changed it, turns out i had an error in my tests (force_blue_labs=force_blue_**saves**), which made me think that for some reason, it doesnt get added as a Normal Door to the dock_weakness list.

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**